airflow/models/baseoperator.py

```diff
@@ -792,6 +854,8 @@ def __setattr__(self, key, value):
         if self._lock_for_execution:
             # Skip any custom behaviour during execute
             return
+        if key in self.__init_kwargs:
```

Comment:
How do we get `self.__init_kwargs` populated ? I could not see it in the change ?

Reply:
In apply_defaults:

```
            if not hasattr(self, '_BaseOperator__init_kwargs'):
                self._BaseOperator__init_kwargs = {}

            result = func(self, **kwargs, default_args=default_args)
            # Store the args passed to init -- we need them to support task.map serialzation!
            self._BaseOperator__init_kwargs.update(kwargs)  # type: ignore
```